azure-mgmt-resources/src/main/java/com/azure/management/BearerTokenAuthenticationPolicy.java

```diff
@@ -0,0 +1,101 @@
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Rewrite the BearerTokenAuthenticationPolicy, it will use default scope when scopes parameter is empty.
+ */
+public class BearerTokenAuthenticationPolicy implements HttpPipelinePolicy {
+    private static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+    private static final String AUTHORIZATION_HEADER_VALUE_FORMAT = "Bearer %s";
+
+    private final Map<String, AccessToken> tokenCache;
+    private final TokenCredential credential;
+    private final String[] scopes;
+    private final AzureEnvironment environment;
+
+    public BearerTokenAuthenticationPolicy(TokenCredential credential, String... scopes) {
+        Objects.requireNonNull(credential);
+        this.credential = credential;
+        this.scopes = scopes;
+        this.tokenCache = new HashMap<>();
+
+        if (credential instanceof AzureTokenCredential) {
+            this.environment = ((AzureTokenCredential) credential).getEnvironment();
+        } else {
+            this.environment = AzureEnvironment.AZURE;
+        }
+    }
+
+    private String getDefaultScopeFromRequest(HttpRequest request) {
+        String host = request.getUrl().getHost();
+        String resource = this.environment.getManagementEndpoint();
+        for (Map.Entry<String, String> endpoint : this.environment.endpoints().entrySet()) {
+            if (host.contains(endpoint.getValue())) {
+                if (endpoint.getKey().equals(AzureEnvironment.Endpoint.KEYVAULT.identifier())) {
+                    resource = String.format("https://%s/", endpoint.getValue().replaceAll("^\\.*", ""));
+                    break;
+                } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.GRAPH.identifier())) {
+                    resource = this.environment.getGraphEndpoint();
+                    break;
+                } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.LOG_ANALYTICS.identifier())) {
+                    resource = this.environment.getLogAnalyticsEndpoint();
+                    break;
+                } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.APPLICATION_INSIGHTS.identifier())) {
+                    resource = this.environment.getApplicationInsightsEndpoint();
+                    break;
+                } else if (endpoint.getKey().equals(AzureEnvironment.Endpoint.DATA_LAKE_STORE.identifier())
+                        || endpoint.getKey().equals(AzureEnvironment.Endpoint.DATA_LAKE_ANALYTICS.identifier())) {
+                    resource = this.environment.getDataLakeEndpointResourceId();
+                    break;
+                }
+            }
+        }
+        return resource + "/.default";
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
```

Comment:
Is it possible to have UPPERCASE of getProtocol?

Reply:
I haven't met a UPPERCASE protocol url, but I think we could handle it.